File: backend/src/main/java/backend/mulkkam/notification/dto/ReadNotificationResponse.java
package backend.mulkkam.notification.dto;

import backend.mulkkam.notification.domain.Notification;
import backend.mulkkam.notification.domain.NotificationType;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

@Schema(description = "개별 알림 응답")
public record ReadNotificationResponse(
        @Schema(description = "알림 ID", example = "1")
        Long id,

        @Schema(description = "알림 제목", example = "물 마실 시간이에요!")
        String title,

        @Schema(description = "알림 타입", example = "REMINDER", implementation = NotificationType.class)
        String type,

        @Schema(description = "알림 생성 시간", example = "2024-01-15T09:30:00")
        LocalDateTime createdAt,

        @Schema(description = "권장 목표량 (ml)", example = "2000", minimum = "0")
        int recommendedTargetAmount,

        @Schema(description = "읽음 여부", example = "false")
        boolean isRead
) {

    public ReadNotificationResponse(Notification notification) {
        this(
                notification.getId(),
                notification.getContent(),
                notification.getNotificationType().name(),
                notification.getCreatedAt(),
                1000, // TODO 2025. 8. 16. 17:19: 수정
                notification.isRead()
        );
    }
}

File: android/app/src/main/java/com/mulkkam/data/remote/model/response/notification/ReadNotificationResponse.kt
package com.mulkkam.data.remote.model.response.notification

import com.mulkkam.domain.model.notification.Notification
import com.mulkkam.domain.model.notification.NotificationType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

@Serializable
data class ReadNotificationResponse(
    @SerialName("type")
    val type: String,
    @SerialName("id")
    val id: Int,
    @SerialName("content")
    val content: String,
    @SerialName("createdAt")
    val createdAt: String,
    @SerialName("isRead")
    val isRead: Boolean,
    @SerialName("recommendedTargetAmount")
    val recommendedTargetAmount: Int? = null,
    @SerialName("applyRecommendAmount")
    val applyRecommendAmount: Boolean? = null,
)

fun ReadNotificationResponse.toDomain(): Notification =
    Notification(
        id = id,
        title = content,
        type = NotificationType.from(type),
        createdAt = LocalDateTime.parse(createdAt, DateTimeFormatter.ISO_DATE_TIME),
        recommendedTargetAmount = recommendedTargetAmount ?: 0,
        isRead = isRead,
        applyRecommendAmount = applyRecommendAmount,
    )
